api/tweets.php

```diff
@@ -21,8 +21,10 @@ function mb_substr_replace($string, $replacement, $start, $length) {
 function getTwitterContent($media_url) {
     $file_name = basename($media_url);
     $info = pathinfo($file_name);
-    $file_path = '../images/twitter/' . $info['filename'] . '.jpg';
+    $file_path = IMAGES_DIR . $info['filename'] . '.jpg';
     if(!file_exists($file_path)) {
+        if (!is_dir(IMAGES_DIR))
+            mkdir(IMAGES_DIR);
```

Comment:
das sollte durch die [.gitignore](https://github.com/FreifunkBremen/bremen.freifunk.net/blob/9e552a02102152777268a4ba43e2a25422dad78e/images/twitter/.gitignore) datei nicht nötig sein, da bei einem git pull automatisch der ordner images/twitter/ angelegt wird

Reply:
Ach so, das mit der `.gitignore` alleine funktioniert so wohl doch nicht; das hätte ich mal dazuschreiben sollen :-)

Der Git-Checkout liegt unter `/home/website/.var/jekyll-envs/bremen.freifunk.net/`. Und der Website-Generator ([Jekyll](https://jekyllrb.com/)) erzeugt aus dem Git-Verzeichnis (wo Markdown-Dateien und so liegen) eine neue Verzeichnis-Struktur mit den HTML-Dateien und Bildern und so; und diese generierte Verzeichnis-Struktur wird dann nach `/var/www/...` kopiert und ist im Browser erreichbar.

Aber Jekyll kopiert anscheinend keine Verzeichnisse da rein, die leer sind oder nur Dot-Files enthalten. Ich hab mal mit den Optionen in `_config.yml` rumgespielt, aber hab das nicht hinbekommen. Und hab dann bemerkt, dass das `/api/cache`-Verzeichnis genau das gleiche Problem hat und wohl deshalb mit `mkdir()` angelegt wird: https://github.com/FreifunkBremen/bremen.freifunk.net/blob/master/api/tweets.php#L38